app/src/main/java/com/wireguard/android/databinding/ObservableKeyedRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright © 2017-2018 WireGuard LLC. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.wireguard.android.databinding
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ObservableList
+import androidx.databinding.ViewDataBinding
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.Adapter
+import com.wireguard.android.BR
+import com.wireguard.android.util.ObservableKeyedList
+import com.wireguard.util.Keyed
+
+import java.lang.ref.WeakReference
+
+/**
+ * A generic `RecyclerView.Adapter` backed by a `ObservableKeyedList`.
+ */
+
+class ObservableKeyedRecyclerViewAdapter<K, E : Keyed<out K>> internal constructor(
+    context: Context, private val layoutId: Int,
+    list: ObservableKeyedList<K, E>
+) : Adapter<ObservableKeyedRecyclerViewAdapter.ViewHolder>() {
+
+    private val callback = OnListChangedCallback(this)
+    private val layoutInflater: LayoutInflater = LayoutInflater.from(context)
+    private var list: ObservableKeyedList<K, E>? = null
+    private var rowConfigurationHandler: RowConfigurationHandler<ViewDataBinding,E>? = null
+
+    init {
+        setList(list)
+    }
+
+    override fun getItemCount(): Int {
+        return if (list != null) list!!.size else 0
```

Comment:
Change this to `list?.size ?: 0`

Reply:
I did this  thing specifically . Will look this afain